models/Duplication_model.slim

```diff
@@ -0,0 +1,167 @@
+initialize() {
+    initializeSLiMOptions(nucleotideBased=T);
+    defineConstant("L", 1010000);              // Genome length
+    defineConstant("DUP_LENGTH", 10000);       // Duplication size
+    defineConstant("DUP_START", L - DUP_LENGTH);  // Start position of duplication
+    defineConstant("N", 500);                  // Population size
+
+    initializeTreeSeq();
+
+    // Generate random nucleotides for the first L - DUP_LENGTH bases
+    randomPart = sample(c("A", "T", "C", "G"), L - DUP_LENGTH - 1, replace=T);
+
+    // Define the segment to duplicate (last DUP_LENGTH bases of randomPart)
+    duplicatedPart = randomPart[(length(randomPart) - DUP_LENGTH):length(randomPart) - 1];
```

Comment:
Does this do exactly what you want it to do?  The tricky thing is that, because of operator precedence, A:B-1 generates a sequence from A-1 to B-1, not a sequence from A to B-1 as people often expect.  If this code does do what you intend it to do, then it would be a good idea to add parentheses to make that clear, as (A:B)-1, so the reader knows it isn't an error.  The parens are unnecessary, and maybe you know what you're doing, but the precendence of : is such a common source of bugs that it is really better to make it very explicitly clear that you're doing what you intend to do.  Check this carefully.